openmetadata-service/src/main/java/org/openmetadata/service/events/subscription/AlertUtil.java

```diff
@@ -218,4 +236,48 @@ public static SubscriptionStatus buildSubscriptionStatus(
         .withNextAttempt(nextAttempt)
         .withTimestamp(timeStamp);
   }
+
+  public static List<ChangeEvent> getFilteredEvent(
+      List<ChangeEvent> events, FilteringRules filteringRules) {
+    List<ChangeEvent> filteredEvents = new ArrayList<>();
+    for (ChangeEvent event : events) {
+      boolean triggerChangeEvent =
+          AlertUtil.shouldTriggerAlert(event.getEntityType(), filteringRules);
+
+      // Evaluate ChangeEvent Alert Filtering
+      if (filteringRules != null
+          && !AlertUtil.evaluateAlertConditions(event, filteringRules.getRules())) {
+        triggerChangeEvent = false;
+      }
+
+      if (triggerChangeEvent) {
+        filteredEvents.add(event);
+      }
+    }
+    return filteredEvents;
+  }
+
+  public static <T> T getValueFromQtzJobMap(JobDataMap dataMap, String key, Class<T> tClass) {
+    Object value = dataMap.get(key);
+    if (value == null) {
+      return null;
+    }
+    return tClass.cast(value);
+  }
+
+  public static EventSubscriptionOffset getInitialAlertOffsetFromDb(UUID eventSubscriptionId) {
+    int eventSubscriptionOffset;
+    String json =
```

Comment:
since we are checking this before, do we need to this load for an offset again?

Reply:
we prevent the call again if this info is present in JobDataMap